Taskfile.yml

```diff
@@ -300,12 +297,25 @@ tasks:
     desc: Build & serve the playground.
     dir: web/playground
     cmds:
-      # Must set `install-links=false` in the playground's `npm install` to
-      # install prql-js as the regular dependency, instead of creating a
-      # symlink. Refer to https://github.com/PRQL/prql/pull/1296.
-      - npm install --install-links=false
+      - task: check-npm-cache
       - npm start
-#
+
+  check-npm-cache:
+    # Look to see if npm dependencies need to be updated
+    # This avoids long start times when no dependencies have changed
+    # Create node_modules/.npmtimestamp after running 'npm install'
+    # Rerun npm install if package.json is newer than the timestamp
+
+    # Must set `install-links=false` in the playground's `npm install` to
+    # install prql-js as the regular dependency, instead of creating a
+    # symlink. Refer to https://github.com/PRQL/prql/pull/1296.
+    desc: Check to see if package.json has changed since we ran npm install
+    dir: web/playground
+    status: # if both are true, don't rebuild
```

Comment:
Can we replace this all with `sources:`? https://taskfile.dev/usage/#by-fingerprinting-locally-generated-files-and-their-sources

Reply:
Probably. My concern is that the `node_modules` folder has ~50,000 files. Computing a checksum over all those files might take nearly as much time as the `npm install` I'll check the`timestamp` method to see how it performs.

Reply:
Can we use it on `package.json` though? Like we're doing here?

Reply:
Remarkably, the checksum option is fast - less than a second. It's also indistinguishable from the `timestamp` method. So this commit switches to using checksums.

My test process was to modify `package.json` (add a blank line) and run `task run-playground` which calls `npm install ...`

I then ^C, and re-ran `task run-playground` and the Task reports, _task: Task "check-npm-cache" is up to date_

Can you suggest any other tests? Thanks.

Reply:
I'm actually not sure if it's using the files under `generates`; I couldn't completely decipher the docs:

> For the checksum (default) or timestamp method to work, it is only necessary to inform the source files. When the timestamp method is used, the last time of the running the task is considered as a generate.